lib/AST/ASTMangler.cpp

```diff
@@ -954,15 +954,15 @@ void ASTMangler::appendType(Type type) {
         fieldsList.push_back(TupleTypeElt(fieldTy, Identifier(), fieldFlag));
       }
       appendTypeList(TupleType::get(fieldsList, tybase->getASTContext())
-                       ->getCanonicalType());
+                       ->getCanonicalType(), /*ignoreElementNames=*/true);
```

Comment:
There are no field names here anyway by construction

Reply:
If there is a field which itself is a tuple type (with field names), then getCanonicalType will flatten the tuple and it will have field names

Reply:
I don't see why creating a new tuple type would be better than passing a flag

Reply:
Because then we don't have to support two modes in the mangler. I'm with Slava on this.

Reply:
That doesn't make sense to me.
Why should I create a new type (which involves memory allocation etc.) instead of just passing a boolean flag?
Maybe it's better to duplicate appendTypeList for both purposes (I though it is not) instead of passing a flag, but creating a temporary type for that is too much overhead.